src/load-glb-and-screenshot.js

```diff
@@ -1,41 +1,32 @@
-const puppeteer = require('puppeteer');
-
-module.exports = async (page, {glbPath, outputPath, format, quality}, timeout) => {
-  return new Promise((resolve) => {
+module.exports = async (page, {glbPath, outputPath, format, quality, timeout}) => {
+  return new Promise((resolve, reject) => {
     page.exposeFunction('resolvePromise', resolve);
-
+    page.exposeFunction('rejectPromise', reject);
     page.evaluate(async (browser_glbPath, browser_outputPath, browser_format, browser_quality, timeout) => {
-      const waitUntil = async (check, interval, timeout) => {
-        var endTime = Number(new Date()) + timeout;
-
-        const checkCondition = (resolve, reject) => {
-          if (check()) {
-            resolve()
-          } else if (Number(new Date()) < endTime) {
-            console.log(`Waiting ${interval}ms for model to render...`);
-            setTimeout(checkCondition, interval, resolve, reject);
-          } else {
-            reject('Wait until timeout');
-          }
+      var endTime = Number(new Date()) + timeout;
+
+      const isTimedOut = function() {
+        const currentTime = Number(new Date())
+        if (currentTime < endTime) {
+          window.logInfo(`Waiting ${endTime - currentTime}ms for model to render...`);
+        }else{
+          window.rejectPromise('Waited until timeout');
         }
-
-        return new Promise(checkCondition);
       }
 
       modelViewer = document.getElementById('snapshot-viewer');
+      modelViewer.addEventListener('model-visibility', function(){
+        const visible = event.detail.visible;
+        if(visible){
+          window.saveDataUrl(
+            modelViewer.toDataURL(browser_format, browser_quality),
```

Comment:
We should consider switching this to use `toBlob` https://github.com/google/model-viewer/pull/890

This will give us better poster images.

Reply:
Created an issue https://github.com/Shopify/screenshot-glb/issues/11